### DISCUSSION TITLE
Frigate/go2rtc remote access with Cloudflare Tunnel & Ngrok

### DISCUSSION THREAD

**Discussion by luisiam** [blakeblackshear/frigate#6144] [Ask A Question] [answered]
I have set up cloudflare tunneling and I am able to remotely access my frigate/go2rtc instance without any issue. The only problem is that webrtc won't work with cloudflare tunneling. I went through the documentation and enabled ngrok in go2rtc. When I try to stream with webrtc remotely, I see this error in the go2rtc log and the stream never starts.

```
11:45:12.412 WRN error="websocket: the client is not using the websocket protocol: 'websocket' token not found in 'Upgrade' header"
```

**Comment by NickM-27** [comment]
when you say you enabled ngrok what do you mean? frigate currently does not include ngrok in the container

**Reply by luisiam** [reply]
I include the ngrok binary inside the /config folder.

```
go2rtc:
  ngrok:
    command: /config/ngrok tcp 8555 --authtoken xxxxxxxxxxxxxxxx
```

**Reply by NickM-27** [reply]
I don't know then, I've not used this setup so not sure what that error would point to. Maybe make an issue on go2rtc

**Reply by luisiam** [reply]
May I ask what kind of configuration you are using to make webrtc work remotely?

**Reply by NickM-27** [reply]
I don't have a symmetrical NAT so I just use udp hole punching

**Comment by luisiam** [comment, marked as answer]
Need to run Frigate docker with `host` network mode and everything works as expected.

**Comment by sammyke007** [comment]
@luisiam can you please explain how you've added ngrok? It's not included bu fedault in Frigate.

**Reply by luisiam** [reply]
I put the ngrok binary under the config directory

**Comment by aaronjrodrigues** [comment]
hi do you mind please sharing your config, im trying to get this setup but for some reason it's not working

**Reply by aaronjrodrigues** [reply]
@luisiam

**Reply by luisiam** [reply]
Everything is here
https://github.com/AlexxIT/go2rtc/issues/367
